hydromt/data_catalog.py

```diff
@@ -1594,6 +1595,85 @@ def get_geodataset(
         )
         return obj
 
+    def get_dataset(
```

Comment:
Not sure if this PR is the right place for it, or if this is a thing for v1.0, but I'm really not a fan of this name, because it implies (to me atleast) that the other dataset kinds like Geodataset etc. inheret from this one. It also does not imply to me anything that the dataset might actually contain, but I do realise it is mostly in line with the other names, so I would be okay with keeping it like this for now, and including this in the larger renaming discussion for 1.0

Reply:
yes, I agree. The name can be a bit confusing. With regards to xarray dataset, I think it is fine the way it is. The get_dataset method will always return a xr.Dataset anyway. 
That being said it is a good idea to take this up in the discussion about names.